game.cc

```diff
@@ -303,20 +354,250 @@ bool Game::isCheck(int player){
 		return false;
 
 	}
-	else return false;
 }
 
-bool Game::makeMove(int startX, int startY, int endX, int endY){
+bool Game::makeMove(int startX, int startY, int endX, int endY, bool promote = true, bool checkForCheck = true){
 	if (isValidMove(startX,startY,endX,endY) == false) return false;
 
+	//Checks to see if piece that it is moving to is own piece
+	for (int i = 0; i < 16; i++){
+		if (currentPlayer() == WHITE){
+			if(playerWhite[i] == theBoard[endX][endY]) return false;
+		}
+		else{
+			if(playerBlack[i] == theBoard[endX][endY]) return false;
+		}
+	}
+
+	//Check for Check
+	if (checkForCheck == true){
+		if (isCheckAfterMove(startX,startY,endX,endY) == true) return false;
+	}
+
+
+	//Checks for Pawn Promotion
+	if (getPawnPromote() && promote == true){
+		//Finds location of piece in the array of pieces
+		int loc;
+		for (int i = 0; i < 16; i++){
+			if (getCurrentPlayer() == WHITE){
+				if(playerWhite[i] == theBoard[startX][startY]){
+					delete playerWhite[i];
+					theBoard[startX][startY] = NULL;
+					loc = i;
+					break;
+				}
+			}
+			else{
+				if(playerBlack[i] == theBoard[startX][startY]){
+					delete playerBlack[i];
```

Comment:
I wonder if instead of deleting the old piece, we just create a new one and replace it's position on the board. Then we can undo if anything goes wrong. The only time we should delete pieces is at the end of the game.